nntrainer/src/databuffer.cpp

```diff
@@ -187,12 +185,15 @@ bool DataBuffer::getDataFromBuffer(BufferType type, vec_4d &outVec,
   switch (type) {
   case BUF_TRAIN: {
     std::vector<int> list;
-    std::unique_lock<std::mutex> ultrain(readyTrainData);
-    cv_train.wait(ultrain, [this]() -> int { return trainReadyFlag; });
-
-    if (train_data.size() < batch_size || trainReadyFlag == DATA_ERROR ||
-        trainReadyFlag == DATA_END) {
-      return false;
+    while (true) {
+      std::unique_lock<std::mutex> ultrain(readyTrainData);
+      cv_train.wait(ultrain, [this]() -> int { return trainReadyFlag; });
```

Comment:
Could you check if this is valid?
https://en.cppreference.com/w/cpp/thread/condition_variable/wait

```suggestion
      cv_train.wait(ultrain, [this]() -> bool{ return trainReadyFlag; });
```

Reply:
fixed. thanks.